### PR TITLE
fix(escrow): correct paisa-to-MATIC conversion — remove 10,000x magnitude error

### LINKED ISSUE #606: [Bug] Escrow deposit converts bid_amount paisa to ETH directly — 10,000x magnitude error on every funded order
### What's broken

In `backend/api/src/routes/orderRoutes.js` at line 646, the escrow deposit amount is computed as:

\`\`\`js
const amountWei = ethers.parseEther((bid.bid_amount / 100).toFixed(2).toString());
\`\`\`

The problem is the unit system. \`bid_amount\` is stored in **paisa** (1 INR = 100 paisa) — this is explicitly documented in \`backend/api/src/lib/pricing.js\`:

> _"All amounts are returned in **paisa** (1 INR = 100 paisa) to match the integer column types... \`load_bids.bid_amount\` is documented as paisa"_

So \`bid.bid_amount / 100\` gives you the value in **INR**. Then \`ethers.parseEther(INR_value)\` treats that INR number as if it were an ETH/MATIC quantity. That's a unit mismatch by definition.

Concrete example: a ₹5,000 freight bid has \`bid_amount = 500000\`. The code does:
- \`500000 / 100 = 5000\`
- \`ethers.parseEther("5000.00")\` → **5,000 MATIC** locked in escrow

At ~$0.40/MATIC, that's $2,000 USD locked for a ₹5,000 (~$60) order. The relayer wallet would need thousands of MATIC to fund real orders. In practice the call either:

1. Reverts with insufficient funds (escrow silently skipped, \`escrow_status\` stays \`pending\` — no on-chain protection for either party), or
2. On a test wallet with large balance, drains it completely

### Code location

- `backend/api/src/routes/orderRoutes.js:646` — the broken conversion
- `backend/api/src/lib/pricing.js:16–20` — documents that \`bid_amount\` is in paisa
- `backend/api/src/validation/requestSchemas.js` — \`submitBidSchema\` confirms \`bid_amount\` is an integer (paisa unit)

### Proposed fix

The conversion needs an explicit INR-to-MATIC exchange rate. Two options:

**Option A (quick fix) — configurable fixed rate via env:**
\`\`\`js
const MATIC_PER_PAISA = parseFloat(process.env.ESCROW_MATIC_PER_PAISA || '0');
if (!MATIC_PER_PAISA) {
  // escrow disabled — log and skip, same as missing wallet addresses
  console.warn('[escrow] ESCROW_MATIC_PER_PAISA not set — skipping deposit.');
} else {
  const maticAmount = (bid.bid_amount * MATIC_PER_PAISA).toFixed(18);
  const amountWei = ethers.parseEther(maticAmount);
  // also add a sanity cap
  const MAX_ESCROW_MATIC = parseFloat(process.env.MAX_ESCROW_MATIC || '5');
  if (parseFloat(maticAmount) > MAX_ESCROW_MATIC) {
    return res.status(400).json({ error: 'Computed escrow amount exceeds safety cap.' });
  }
}
\`\`\`

**Option B (correct long-term) — real-time price oracle:**
Fetch MATIC/INR rate from Chainlink or a price API at bid acceptance time, then compute `(bid_amount_in_INR / matic_price_inr)` as the MATIC amount.

At minimum, the safety cap guard should go in regardless of which approach is chosen, so a unit bug never drains a wallet silently.

### PR DESCRIPTION
**Pull request by nyxsky404**
Closes #606

## What was broken

`bid_amount` is stored in paisa (1 INR = 100 paisa, documented in `lib/pricing.js`). The accept-bid handler was doing:

```js
const amountWei = ethers.parseEther((bid.bid_amount / 100).toFixed(2).toString());
```

`bid_amount / 100` converts paisa to INR. Then `ethers.parseEther(INR_value)` treats that number as MATIC. A ₹5,000 bid (`bid_amount=500000`) would lock **5,000 MATIC** (~$2,000) on-chain for a $60 order.

## Fix

Replaced with a configurable `ESCROW_MATIC_PER_PAISA` env var that holds the explicit INR→MATIC exchange rate at the paisa granularity. If the var is not set the deposit is skipped with a warning (same existing behaviour as a missing wallet address). A `MAX_ESCROW_MATIC` cap rejects any deposit above the configured ceiling before it reaches the chain.

Both variables are documented in `.env.example`.

## Summary by CodeRabbit

* **New Features**
  * Added configurable on-chain escrow deposit support for MATIC, including an environment-defined conversion rate and a maximum deposit cap. Leaving the conversion rate unset disables on-chain escrow deposits.

* **Bug Fixes**
  * Updated escrow deposit calculation during bid acceptance to use the configured conversion rate instead of a fixed formula.
  * Added validation for the maximum allowed MATIC amount, rejecting requests that exceed the cap and only proceeding to escrow transaction creation when inputs are valid.

### CODE REVIEW

File: backend/api/.env.example [hunk 1]
```diff
@@ -35,3 +35,11 @@ POLYGON_RPC_URL=https://polygon-rpc.com
 REPUTATION_CONTRACT_ADDRESS=0x...
 DELIVERY_RECEIPTS_CONTRACT_ADDRESS=0x...
 RELAYER_WALLET_PRIVATE_KEY=your_private_key
+
+# ── Escrow ────────────────────────────────
+# MATIC equivalent of 1 paisa. bid_amount is stored in paisa (1 INR = 100 paisa).
+# Example: if 1 INR = 0.00040 MATIC, set ESCROW_MATIC_PER_PAISA=0.000004
+# Leave unset to disable on-chain escrow deposits.
+ESCROW_MATIC_PER_PAISA=0.000004
```
[the remark this answers is not in the document]
Comment (nyxsky404) [reply backend/api/.env.example]: Fixed in 5d36f1d. Cleared the default value to `ESCROW_MATIC_PER_PAISA=` so teams copying the file don't accidentally enable live deposits with a stale rate.

File: backend/api/src/routes/orderRoutes.js [hunk 2]
```diff
@@ -669,23 +669,33 @@ router.post('/:id/bids/:bidId/accept', authenticate, requireRole(['customer']),
...
-        } else {
+      const maticPerPaisa = parseFloat(process.env.ESCROW_MATIC_PER_PAISA || '0');
+      if (!maticPerPaisa || !isFinite(maticPerPaisa) || maticPerPaisa <= 0) {
+        logger.warn('[escrow] ESCROW_MATIC_PER_PAISA not configured — skipping escrow deposit.');
+      } else {
+        const maticAmount = (bid.bid_amount * maticPerPaisa).toFixed(18);
+        const maxEscrowMatic = parseFloat(process.env.MAX_ESCROW_MATIC || '5');
+        if (parseFloat(maticAmount) > maxEscrowMatic) {
+          return res.status(400).json({ error: 'Computed escrow amount exceeds safety cap. Check ESCROW_MATIC_PER_PAISA configuration.' });
+        }
```
[the remark this answers is not in the document]
Comment (nyxsky404) [reply backend/api/src/routes/orderRoutes.js]: Fixed in 5d36f1d. Added explicit `Number.isFinite` + positive check on the parsed value — if it's invalid we now fail closed with 500 rather than silently dropping the cap.

File: backend/api/src/routes/orderRoutes.js [hunk 3]
```diff
@@ -669,23 +669,33 @@ router.post('/:id/bids/:bidId/accept', authenticate, requireRole(['customer']),
...
+        const amountWei = ethers.parseEther(maticAmount);
```
[the remark this answers is not in the document]
Comment (nyxsky404) [reply backend/api/src/routes/orderRoutes.js:692]: The concrete failure mode you flagged (NaN disabling the cap) is fixed in 5d36f1d with an explicit `Number.isFinite` guard on `MAX_ESCROW_MATIC`.

On the bigint suggestion: `toFixed(18)` goes directly into `ethers.parseEther()` which does the string→wei conversion internally — no further float arithmetic runs after that point. The rate is operator-supplied with arbitrary decimal precision, so a string-based parseEther boundary is appropriate here. Happy to revisit if there's a specific precision edge case you have in mind.

**Comment by nyxsky404** [comment]
CI failures are pre-existing — backend test failures (expected 500, got 422) exist on `main` before this PR due to the wallet-address 422 guard firing before the RPC call in those test setups. Flutter failures are unrelated to this change.
